azurerm/internal/services/network/virtual_network_gateway_connection_resource.go

```diff
@@ -326,6 +329,9 @@ func resourceVirtualNetworkGatewayConnectionCreateUpdate(d *schema.ResourceData,
 		VirtualNetworkGatewayConnectionPropertiesFormat: properties,
 	}
 
+	locks.ByName(*properties.VirtualNetworkGateway1.Name, virtualNetworkGatewayConnectionResourceName)
```

Comment:
Shall we use the `locks.ByID` to lock against the ID of the including vnet gw instead? Otherwise, it might unecessarily locks two same named connections in different resource groups?